src/ResourceManager/IotHub/Commands.IotHub/Models/PSRoutingServiceBusQueueEndpointProperties.cs

```diff
@@ -0,0 +1,95 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Management.IotHub.Models
+{
+    using System;
+    using System.Linq;
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+    using Microsoft.Rest;
+    using Microsoft.Rest.Serialization;
+    using Microsoft.Rest.Azure;
+
+    /// <summary>
+    /// The properties related to service bus queue endpoint types.
+    /// </summary>
+    public partial class PSRoutingServiceBusQueueEndpointProperties
```

Comment:
What is the rationale for having a separate class for this?  Isn't there a more generic Routing Proeprties that will do?